src/lib/array/timsort.ml

```diff
@@ -63,63 +63,86 @@ let%test_module _ =
 
 let rec merge_lo
   cmp
-  dest ofs
-  src0 ofs0 len0
-  src1 ofs1 len1
+  dest beg
+  src0 beg0 end0 x0
+  src1 beg1 end1 x1
```

Comment:
I think it would be worth having comments like `(* x0 = src0.(beg0) *)` already at this point. It's readable in the assertions below but I think it's better to have it here!